crypto/cmp/cmp_msg.c

```diff
@@ -994,3 +980,14 @@ OSSL_CMP_MSG *ossl_cmp_msg_load(const char *file)
     BIO_free(bio);
     return msg;
 }
+
+OSSL_CMP_MSG *OSSL_d2i_CMP_MSG_bio(BIO *bio, OSSL_CMP_MSG **msg)
+{
+    return ASN1_d2i_bio_of(OSSL_CMP_MSG, OSSL_CMP_MSG_new,
+                           d2i_OSSL_CMP_MSG, bio, msg);
+}
+
+int OSSL_i2d_CMP_MSG_bio(BIO *bio, const OSSL_CMP_MSG *msg)
+{
+    return ASN1_i2d_bio_of(OSSL_CMP_MSG, i2d_OSSL_CMP_MSG, bio, msg);
+}
```

Comment:
Things you don't notice before you get a conflict in `util/libcrypto.num`...

These lines happened to catch my eyes:

```
OSSL_d2i_CMP_MSG_bio                    ?	3_0_0	EXIST::FUNCTION:CMP
OSSL_i2d_CMP_MSG_bio                    ?	3_0_0	EXIST::FUNCTION:CMP
```

That was kind of an odd name, when we basically already have established the pattern `i2d_TYPE_bio` and `d2i_TYPE_bio`...  and especially considering that there are already these (automatically defined with ASN1 constructor macros):

```
d2i_OSSL_CMP_MSG                        ?	3_0_0	EXIST::FUNCTION:CMP
i2d_OSSL_CMP_MSG                        ?	3_0_0	EXIST::FUNCTION:CMP
```

Essentially, the `i2d_` and `d2i_` pattern is already established, has been for a long time, so please don't re-invent it.

I suggest a name fix in a separate PR.

Reply:
Oh....good point. I missed this!

Reply:
Good point; done in https://github.com/openssl/openssl/pull/11307.